File: csharp/Kobiton/Configs.cs
using OpenQA.Selenium.Remote;
using System;
using System.Collections.Generic;
using System.Text;

namespace Kobiton
{
  public class Configs
  {

    private const string AUTH_USERNAME = "";
    private const string AUTH_ACCESSKEY = "";
    public const int SESSION_TIMEOUT = 20*60;

    public static Uri kobitonServerUrl()
    {
      return new Uri("https://api.kobiton.com/wd/hub");
    }

    public static DesiredCapabilities desiredCapabilitiesAndroidWeb()
    {
      DesiredCapabilities capabilities = new DesiredCapabilities();
      capabilities.SetCapability("username", AUTH_USERNAME);
      capabilities.SetCapability("accessKey", AUTH_ACCESSKEY);
      capabilities.SetCapability("sessionName", "Android Web");
      capabilities.SetCapability("sessionDescription", "This is an example for Android Web testing");
      capabilities.SetCapability("deviceOrientation", "portrait");
      capabilities.SetCapability("captureScreenshots", true);
      capabilities.SetCapability("browserName", "chrome");
      capabilities.SetCapability("deviceGroup", "KOBITON");
      capabilities.SetCapability("deviceName", "Galaxy S5");
      capabilities.SetCapability("platformName", "Android");
      return capabilities;
    }

    public static DesiredCapabilities desiredCapabilitiesAndroidApp()
    {
      DesiredCapabilities capabilities = new DesiredCapabilities();
      capabilities.SetCapability("username", AUTH_USERNAME);
      capabilities.SetCapability("accessKey", AUTH_ACCESSKEY);
      capabilities.SetCapability("sessionName", "Android app");
      capabilities.SetCapability("sessionDescription", "This is an example for Android app testing");
      capabilities.SetCapability("deviceOrientation", "portrait");
      capabilities.SetCapability("captureScreenshots", true);
      capabilities.SetCapability("app", "https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/iFixit.apk");
      capabilities.SetCapability("deviceGroup", "KOBITON");
      capabilities.SetCapability("deviceName", "Galaxy S5");
      capabilities.SetCapability("platformName", "Android");
      return capabilities;
    }

    public static DesiredCapabilities desiredCapabilitiesiOSWeb()
    {
      DesiredCapabilities capabilities = new DesiredCapabilities();
      capabilities.SetCapability("username", AUTH_USERNAME);
      capabilities.SetCapability("accessKey", AUTH_ACCESSKEY);
      capabilities.SetCapability("sessionName", "iOS Web");
      capabilities.SetCapability("sessionDescription", "This is an example for iOS Web testing");
      capabilities.SetCapability("deviceOrientation", "portrait");
      capabilities.SetCapability("captureScreenshots", true);
      capabilities.SetCapability("browserName", "safari");
      capabilities.SetCapability("deviceGroup", "KOBITON");
      capabilities.SetCapability("deviceName", "iPhone 7");
      capabilities.SetCapability("platformName", "iOS");
      return capabilities;
    }

    public static DesiredCapabilities desiredCapabilitiesiOSApp()
    {
      DesiredCapabilities capabilities = new DesiredCapabilities();
      capabilities.SetCapability("username", AUTH_USERNAME);
      capabilities.SetCapability("accessKey", AUTH_ACCESSKEY);
      capabilities.SetCapability("sessionName", "iOS app");
      capabilities.SetCapability("sessionDescription", "This is an example for iOS App testing");
      capabilities.SetCapability("deviceOrientation", "portrait");
      capabilities.SetCapability("captureScreenshots", true);
      capabilities.SetCapability("app", "https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/iFixit.ipa");
      capabilities.SetCapability("deviceGroup", "KOBITON");
      capabilities.SetCapability("deviceName", "iPhone 7");
      capabilities.SetCapability("platformName", "iOS");
      return capabilities;
    }
  }
}

File: java/java_testng_junit/src/main/java/kobiton/com/configs/Configs.java
package kobiton.com.configs;

import java.net.MalformedURLException;
import java.net.URL;

import org.apache.commons.codec.binary.Base64;
import org.openqa.selenium.remote.DesiredCapabilities;

public class Configs {

    static String USERNAME = "";
    static String ACCESS_KEY = "";
    public static final URL kobitonServerUrl(){
        try {
            return new URL("https://" + USERNAME + ":" + ACCESS_KEY + "@api.kobiton.com/wd/hub");
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
        return null;
    }

    public static String generateBasicAuth(String username, String apiKey) {
        String authString = username + ":" + apiKey;
        byte[] authEncBytes = Base64.encodeBase64(authString.getBytes());
        String authStringEnc = new String(authEncBytes);
        return "Basic " + authStringEnc;
    }
    
    public static DesiredCapabilities desiredCapabilitiesAndroidWeb(){
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability("sessionName", "Android Web");
        capabilities.setCapability("sessionDescription", "This is an example for Android Web testing");
        capabilities.setCapability("deviceOrientation", "portrait");
        capabilities.setCapability("captureScreenshots", true);
        capabilities.setCapability("browserName", "chrome");
        capabilities.setCapability("deviceGroup", "KOBITON");
        capabilities.setCapability("deviceName", "Galaxy S5");
        capabilities.setCapability("platformName", "Android");
        return capabilities;
    }

    public static DesiredCapabilities desiredCapabilitiesAndroidApp(){
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability("sessionName", "Android app");
        capabilities.setCapability("sessionDescription", "This is an example for Android app testing");
        capabilities.setCapability("deviceOrientation", "portrait");
        capabilities.setCapability("captureScreenshots", true);
        capabilities.setCapability("app", "https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/com.dozuki.ifixit.apk");
        capabilities.setCapability("deviceGroup", "KOBITON");
        capabilities.setCapability("deviceName", "Galaxy S5");
        capabilities.setCapability("platformName", "Android");
        capabilities.setCapability("allowMobileShell", true);
        return capabilities;
    }

    public static DesiredCapabilities desiredCapabilitiesiOSWeb(){
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability("sessionName", "iOS Web");
        capabilities.setCapability("sessionDescription", "This is an example for iOS Web testing");
        capabilities.setCapability("deviceOrientation", "portrait");
        capabilities.setCapability("captureScreenshots", true);
        capabilities.setCapability("browserName", "safari");
        capabilities.setCapability("deviceGroup", "KOBITON");
        capabilities.setCapability("deviceName", "iPhone 7");
        capabilities.setCapability("platformName", "iOS");
        return capabilities;
    }

    public static DesiredCapabilities desiredCapabilitiesiOSApp(){
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability("sessionName", "iOS app");
        capabilities.setCapability("sessionDescription", "This is an example for iOS App testing");
        capabilities.setCapability("deviceOrientation", "portrait");
        capabilities.setCapability("captureScreenshots", true);
        capabilities.setCapability("app", "https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/iFixit.ipa");
        capabilities.setCapability("deviceGroup", "KOBITON");
        capabilities.setCapability("deviceName", "iPhone 7");
        capabilities.setCapability("platformName", "iOS");
        return capabilities;
    }
}
